tasks/main.yml

```diff
@@ -113,52 +113,51 @@
         name: mssql-server
         state: started
 
+    - name: Check if a custom tcpport setting exist
+      command: /opt/mssql/bin/mssql-conf get network tcpport
+      ignore_errors: true
+      changed_when: false
+      register: __mssql_custom_tcp_port
+
+    - name: Check if a custom ipaddress setting exist
+      command: /opt/mssql/bin/mssql-conf get network ipaddress
+      ignore_errors: true
+      changed_when: false
+      register: __mssql_custom_ip_address
+
     - name: Check if the set password matches the existing password
-      shell: |
-        tcpport_output=$(/opt/mssql/bin/mssql-conf get network tcpport)
-        tcpport=$(echo ${tcpport_output} | sed 's/tcpport : //')
-        ipaddress_output=$(/opt/mssql/bin/mssql-conf get network ipaddress)
-        ipaddress=$(echo ${ipaddress_output} | sed 's/ipaddress : //')
-        if [[ ${tcpport_output} == tcpport* ]] &&
-        [[ ${ipaddress_output} == ipaddress* ]]; then
-          password_query="$(/opt/mssql-tools/bin/sqlcmd \
-          -S ${ipaddress},${tcpport} -U sa -P '{{ mssql_password }}' \
-          -Q 'SELECT @@VERSION')"
-        elif [[ ${tcpport_output} == tcpport* ]] &&
-        [[ ${ipaddress_output} != ipaddress* ]]; then
-          password_query="$(/opt/mssql-tools/bin/sqlcmd \
-          -S localhost,${tcpport} -U sa -P '{{ mssql_password }}' \
-          -Q 'SELECT @@VERSION')"
-        elif [[ ${tcpport_output} != tcpport* ]] &&
-        [[ ${ipaddress_output} == ipaddress* ]]; then
-          password_query="$(/opt/mssql-tools/bin/sqlcmd \
-          -S ${ipaddress} -U sa -P '{{ mssql_password }}' \
-          -Q 'SELECT @@VERSION')"
-        else
-          password_query="$(/opt/mssql-tools/bin/sqlcmd \
-          -U sa -P '{{ mssql_password }}' -Q 'SELECT @@VERSION')"
-        fi
-        if [[ ${password_query} == *"Microsoft SQL Server"* ]]; then
-          echo true
-        else
-          echo false
-        fi
+      vars:
+        __ipaddress: >-
+          "{{ __mssql_custom_ip_address.stdout | regex_search('[0-9].*') }}"
+        __tcpport: >-
+          "{{ __mssql_custom_tcp_port.stdout | regex_search('[0-9].*') }}"
+        __s_arg: >-
+          "{{ (__ipaddress or __tcpport) | ternary('-S', '') }}"
+        __ipaddress_arg: >-
+          "{{ __ipaddress if __ipaddress else
+          'localhost' if __tcpport else '' }}"
+      command: >-
+        /opt/mssql-tools/bin/sqlcmd
+        {{ __s_arg }}
+        {{ __ipaddress_arg }}{{ ',' if __tcpport
+        else '' }}{{ __tcpport if __tcpport else '' }}
+        -U sa -P {{ mssql_password | quote }} -Q 'SELECT @@VERSION'
       ignore_errors: true
       changed_when: false
-      register: __mssql_password_matches
+      register: __mssql_password_query
 
     - name: Ensure that the mssql-server service is stopped
       service:
         name: mssql-server
         state: stopped
-      when: not __mssql_password_matches.stdout | bool
+      when: __mssql_password_query.rc != 0
```

Comment:
to test a registered variable for command success/failure use the test https://docs.ansible.com/ansible/2.8/user_guide/playbooks_tests.html#task-results `when: __mssql_password_query is failed`